_bleio/scan_entry.py

```diff
@@ -0,0 +1,155 @@
+# The MIT License (MIT)
+#
+# Copyright (c) 2020 Dan Halbert for Adafruit Industries LLC
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+"""
+`_bleio.scan_entry`
+=======================================================================
+
+_bleio implementation for Adafruit_Blinka_bleio
+
+* Author(s): Dan Halbert for Adafruit Industries
+"""
+
+from _bleio import Address, UUID
+
+
+class ScanEntry:
+    def __init__(self, device):
+        self._device = device
+        self._address = Address(string=device.address)
+        self._data_dict = self._build_data_dict()
+
+    def matches(
+        self, prefixes, all: bool = True  # pylint: disable=redefined-builtin
+    ) -> bool:
+        # We don't have the original advertisement bytes, so we can't
+        # do a perfect job of matching.
+        if len(prefixes) == 0:
+            return True
+        fields = self.advertisement_fields
+        for prefix in self._separate_prefixes(prefixes):
+            prefix_matched = False
+            for field in fields:
+                if field.startswith(prefix):
+                    if not all:
+                        return True
+                    prefix_matched = True
+                    break
+            # if all, this prefix must match at least one field
+            if not prefix_matched and all:
+                return False
+
+        # All prefixes matched some field (if all), or none did (if any).
+        return all
+
+    def __repr__(self):
+        return str(self)
+
+    def __str__(self):
+        return str(self._device)
+
+    @property
+    def address(self):
+        return self._address
+
+    @property
+    def rssi(self):
+        return self._device.rssi
+
+    @property
+    def connectable(self):
+        """This is unknown on most Blinka implementations. Always returns ``None``."""
+        return None
```

Comment:
Any concern about None evaluating to False? Most scans are probably connectable.

Reply:
This is hard. Maybe it could be a better guess based on what's in the advertisement.

Reply:
True. It's likely if any service related ADTs are included.

Reply:
Now I check for services in the advertisement and guess that it means it's connectable.